### PR TITLE
Pick getLockByKey to master

### LINKED ISSUE #8697: Adaption works for cse tiflash

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #8697

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/Storages/KVStore/MultiRaft/RegionData.cpp [hunk 1]
```diff
@@ -262,6 +262,28 @@ DecodedLockCFValuePtr RegionData::getLockInfo(const RegionLockReadQuery & query)
...
+        const auto & [tikv_key, tikv_val, lock_info_ptr] = lock_it->second;
+        std::ignore = tikv_key;
+        std::ignore = lock_info_ptr;
+        return tikv_val;
+    }
+
+    // It is safe to ignore the missing lock key after restart, print a warning log and return nullptr
+    LOG_WARNING(
+        Logger::get(),
+        "Failed to get lock by key {} in region data, map size {}, count {}",
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/MultiRaft/RegionData.cpp:280]:
```suggestion
        "Failed to get lock by key in region data, key={} map_size={} count={}",
```

File: dbms/src/Storages/KVStore/tests/gtest_new_kvstore.cpp [hunk 2]
```diff
@@ -27,6 +27,37 @@ namespace DB
...
+        auto [index, term] = proxy_instance->rawWrite(
+            1,
+            {str_key, str_key},
+            {str_lock_value, str_val_default},
+            {WriteCmdType::Put, WriteCmdType::Put},
+            {ColumnFamilyType::Lock, ColumnFamilyType::Default});
+        UNUSED(term);
+        proxy_instance->doApply(kvs, ctx.getTMTContext(), cond, 1, index);
+        ASSERT_EQ(kvr1->getLockByKey(str_key)->dataSize(), str_lock_value.size());
+    }
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/tests/gtest_new_kvstore.cpp:56]: Also add a test for getting a lock by non-existent key return nullptr

**Pull request merged**